doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,115 +1257,87 @@ Managing the local context
    for any unresolved variables rather than failing.  Typically this happens when
    using a ``%`` intropattern (see :n:`@simple_intropattern`).
 
-.. tacn:: clear @ident
-   :name: clear
+.. tacn:: clear {? {? - } {+ @ident } }
 
-   This tactic erases the hypothesis named :n:`@ident` in the local context of
-   the current goal. As a consequence, :n:`@ident` is no more displayed and no
-   more usable in the proof development.
+   clear
+   clear ident ...
+   clear - ident ...
 
-   .. exn:: No such hypothesis.
-      :undocumented:
+   Erases hypotheses from the context of the current goal, provided that the
+   remaining hypotheses and the goal don't depend directly or indirectly on the
+   erased hypotheses.
+   That means the hypotheses will no longer appear in the context and therefore
+   can't be used in subsequent proof steps.
+
+   :n:`clear`
+     All hypotheses are erased.  This may leave the context unchanged; this form never fails.
+
+   :n:`clear {+ @ident }`
+     Erases the named hypotheses.
+
+   :n:`clear - {+ @ident }`
+     Erases all hypotheses that are not named by(ass) the :n:`@ident`\s.
```

Comment:
What is "(ass)"?

It seems to me that the previous description was clearer and should be reused / adapted:

>This variant clears all the hypotheses except the ones depending in the hypotheses named :n:`{+ @ident}` and in the goal.

In particular, this form too can leave the goal unchanged and will never fail.

Reply:
> What is "(ass)"?

Pretty funny??  A vulgar anatomical term?  I have no idea.  Maybe my cat walking on the keyboard?  Except I don't have a cat.

> In particular, this form too can leave the goal unchanged and will never fail.

It can fail if there's no hypothesis matching `@ident`.